src/test/com/fulcrologic/fulcro/routing/legacy_ui_routers_spec.cljc

```diff
@@ -18,10 +19,24 @@
    :router-targets {:PAGE/simple-target SimpleTarget}
    :router-id      :PAGE/root-router})
 
+(comp/defsc Root [this _]
+          {:query         [{:root-router (comp/get-query SimpleTarget)}]
+           :initial-state (fn [& _]
+                            {:root-router (comp/get-initial-state SimpleTarget _)})})
+
 (specification "defsc-router Macro"
   (component "Basic feature access"
     (assertions
       "Just returns it's query"
       (comp/get-query SimpleRouter)
       => [::fr/id
-          {::fr/current-route {:PAGE/simple-target [:PAGE/id :PAGE/ident]}}])))
+          {::fr/current-route {:PAGE/simple-target [:PAGE/id :PAGE/ident]}}]))
+  (component "Normalization"
```

Comment:
So, I would prefer tests that only involve the items under test.  In the case you're reporting, a compile error will result if the macro is incorrect.  Therefore, simple defining a router should reproduce the error.  If you want to add a test of something, then a test like: "router initial state can be defined and read" where you've defined it, allow it to take params, USE the params in initial state, and verify that the map that comes back uses the param.

Drop the whole `tree->db` stuff...it is noise and makes the test harder to understand.